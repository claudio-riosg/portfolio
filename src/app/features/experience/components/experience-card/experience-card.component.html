<div class="experience-card">
  <!-- Header with company and role info -->
  <div 
    class="experience-header"
    appScrollReveal
    [threshold]="0.3"
    [delay]="50"
  >
    <h3 class="company">{{ experience().company }}</h3>
    <div class="role-period">
      <span class="role">{{ experience().title }}</span>
      <span class="period">({{ experience().period }})</span>
    </div>
  </div>
  
  <!-- Summary section -->
  <p 
    class="summary"
    appScrollReveal
    [threshold]="0.4"
    [delay]="100"
  >
    {{ experience().summary }}
  </p>
  
  <!-- Details list with staggered animation -->
  <ul class="details">
    @for (detail of experience().details; track $index; let i = $index) {
      <li
        appScrollReveal
        [threshold]="0.5"
        [delay]="150 + (i * 50)"
        [staggerDelay]="50"
      >
        {{ detail }}
      </li>
    }
  </ul>
</div> 