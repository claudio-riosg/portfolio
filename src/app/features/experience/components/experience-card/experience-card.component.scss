.experience-card {
  overflow: hidden;
  position: relative;
  background: var(--card-bg);
  backdrop-filter: blur(20px);
  border: 1px solid var(--card-border-color);
  border-radius: var(--card-border-radius);
  padding: var(--space-xl);
  margin-bottom: var(--space-xl);
  box-shadow: var(--shadow-card-default);
  transition: all var(--transition-duration-lg) var(--transition-easing-standard);
  transform-style: preserve-3d;
  
  &:hover {
    box-shadow: var(--shadow-card-hover);
    border-color: var(--card-border-color-hover);
    background: var(--card-bg-hover);
    
    .experience-header .company {
      background: var(--gradient-text-hero);
      -webkit-background-clip: text;
      -webkit-text-fill-color: transparent;
      background-clip: text;
    }
  }

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 4px;
    background: linear-gradient(90deg, 
      var(--color-primary) 0%, 
      var(--color-secondary) 50%,
      var(--color-accent) 100%);
    border-radius: var(--card-border-radius) var(--card-border-radius) 0 0;
    opacity: 0;
    transition: opacity var(--transition-duration-md) ease;
    pointer-events: none;
    z-index: 1;
  }

  &:hover::before {
    opacity: 1;
  }

  .experience-header {
    margin-bottom: var(--space-lg);
    position: relative;
    
    .company {
      font-size: var(--font-size-xl);
      font-weight: var(--font-weight-bold);
      margin: 0 0 var(--space-sm) 0;
      color: var(--color-heading);
      line-height: 1.3;
      letter-spacing: -0.02em;
      transition: all var(--transition-duration-md) var(--transition-easing-standard);
      position: relative;
      
      &::after {
        content: '';
        position: absolute;
        bottom: -8px;
        left: 0;
        width: 0;
        height: 2px;
        background: linear-gradient(90deg, 
          var(--color-primary), 
          var(--color-accent));
        border-radius: 1px;
        transition: width var(--transition-duration-lg) var(--transition-easing-standard);
      }
    }

    .role-period {
      display: flex;
      flex-direction: column;
      gap: var(--space-xs);
      
      @media (min-width: 768px) {
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
      }
      
      .role {
        font-size: var(--font-size-md);
        font-weight: var(--font-weight-semibold);
        color: var(--color-primary);
        text-transform: uppercase;
        letter-spacing: 0.5px;
        position: relative;
        display: inline-block;
        
        &::before {
          content: '';
          position: absolute;
          top: 50%;
          left: -20px;
          transform: translateY(-50%);
          width: 12px;
          height: 2px;
          background: var(--color-primary);
          border-radius: 1px;
        }
      }
      
      .period {
        font-size: var(--font-size-base);
        font-weight: var(--font-weight-normal);
        color: var(--color-text-secondary);
        opacity: 0.8;
        font-style: italic;
      }
    }
  }

  .summary {
    font-size: var(--font-size-md);
    line-height: 1.7;
    color: var(--color-text-primary);
    margin: 0 0 var(--space-xl) 0;
    font-weight: var(--font-weight-normal);
    position: relative;
    padding-left: var(--space-lg);
    
    &::before {
      content: '"';
      position: absolute;
      left: 0;
      top: -5px;
      font-size: 2.5rem;
      color: var(--color-primary);
      opacity: 0.3;
      font-family: Georgia, serif;
      line-height: 1;
    }
  }

  .details {
    list-style: none;
    padding: 0;
    margin: 0;
    
    li {
      position: relative;
      padding: var(--space-sm) 0 var(--space-sm) var(--space-xl);
      margin-bottom: var(--space-xs);
      font-size: var(--font-size-base);
      line-height: 1.6;
      color: var(--color-text-primary);
      border-left: 2px solid transparent;
      transition: all var(--transition-duration-md) ease;
      background: linear-gradient(135deg, 
        rgba(var(--color-primary-rgb), 0.02) 0%, 
        rgba(var(--color-accent-rgb), 0.02) 100%);
      border-radius: 8px;
      
      &::before {
        content: '';
        position: absolute;
        left: var(--space-sm);
        top: 50%;
        transform: translateY(-50%);
        width: 6px;
        height: 6px;
        background: var(--color-primary);
        border-radius: 50%;
        opacity: 0.6;
        transition: all var(--transition-duration-md) ease;
      }
      
      &:hover {
        padding-left: calc(var(--space-xl) + var(--space-xs));
        border-left-color: var(--color-primary);
        background: linear-gradient(135deg, 
          rgba(var(--color-primary-rgb), 0.05) 0%, 
          rgba(var(--color-accent-rgb), 0.05) 100%);
        
        &::before {
          opacity: 1;
          transform: translateY(-50%) scale(1.3);
          background: var(--color-accent);
        }
      }
      
      &:last-child {
        margin-bottom: 0;
      }
    }
  }
}

// Dark theme support
:global(.dark-theme) .experience-card {
  background: linear-gradient(135deg, 
    rgba(0, 0, 0, 0.2) 0%, 
    rgba(0, 0, 0, 0.1) 100%);
  border-color: rgba(255, 255, 255, 0.1);
  
  &:hover {
    border-color: rgba(255, 255, 255, 0.2);
  }
  
  .experience-header {
    .company {
      color: var(--text-primary-dark, #f9fafb);
    }
    
    .role-period {
      .period {
        color: var(--text-secondary-dark, #9ca3af);
      }
    }
  }
  
  .summary {
    color: var(--text-primary-dark, #e5e7eb);
  }
  
  .details li {
    color: var(--text-primary-dark, #d1d5db);
  }
}

// Animation for card hover on company name underline
.experience-card:hover .experience-header .company::after {
  width: 100%;
}

// Responsive design
@media (max-width: 768px) {
  .experience-card {
    padding: var(--space-lg);
    margin-bottom: var(--space-lg);
    
    .experience-header {
      .company {
        font-size: var(--font-size-lg);
      }
      
      .role-period {
        .role {
          font-size: var(--font-size-base);
          
          &::before {
            left: -15px;
            width: 10px;
          }
        }
      }
    }
    
    .summary {
      font-size: var(--font-size-base);
      padding-left: var(--space-md);
      
      &::before {
        font-size: 2rem;
      }
    }
    
    .details li {
      padding-left: var(--space-lg);
      font-size: var(--font-size-sm);
      
      &:hover {
        padding-left: calc(var(--space-lg) + var(--space-xs));
      }
    }
  }
}

@media (max-width: 1024px) {
  .experience-card {
    max-width: 90vw;
  }
}

@media (max-width: 480px) {
  .experience-card {
    padding: var(--space-sm, 1rem) var(--space-xs, 0.5rem);
    max-width: 100vw;
  }
}

@media (max-width: 600px) {
  .experience-card {
    padding: var(--space-md);
    border-radius: 12px;
    box-shadow: 0 2px 8px rgba(0,0,0,0.08);
  }
} 