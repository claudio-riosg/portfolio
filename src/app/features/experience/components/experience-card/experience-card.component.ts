import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WorkExperience } from '@app/core/models/work-experience.interface';

@Component({
  selector: 'app-experience-card',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './experience-card.component.html',
  styleUrl: './experience-card.component.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ExperienceCardComponent {
  @Input({ required: true }) experience!: WorkExperience;
} 