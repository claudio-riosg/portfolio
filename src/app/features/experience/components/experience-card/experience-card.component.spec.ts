import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ExperienceCardComponent } from './experience-card.component';
import { By } from '@angular/platform-browser';
import { WorkExperience } from '@app/core/models/work-experience.interface';

describe('ExperienceCardComponent', () => {
  let component: ExperienceCardComponent;
  let fixture: ComponentFixture<ExperienceCardComponent>;

  const mockExperience: WorkExperience = {
    company: 'Test Company',
    title: 'Test Title',
    period: '2020 - 2022',
    summary: 'Test summary for the experience card.',
    details: [
      'First detail',
      'Second detail',
      'Third detail',
    ],
  };

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ExperienceCardComponent],
    }).compileComponents();
    fixture = TestBed.createComponent(ExperienceCardComponent);
    component = fixture.componentInstance;
    component.experience = mockExperience;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render company, title, period, and summary', () => {
    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled.querySelector('.company')?.textContent).toContain('Test Company');
    expect(compiled.querySelector('.role')?.textContent).toContain('Test Title');
    expect(compiled.querySelector('.period')?.textContent).toContain('2020 - 2022');
    expect(compiled.querySelector('.summary')?.textContent).toContain('Test summary');
  });

  it('should render all details as list items', () => {
    const items = fixture.debugElement.queryAll(By.css('.details li'));
    expect(items.length).toBe(mockExperience.details.length);
    items.forEach((item, idx) => {
      expect(item.nativeElement.textContent).toContain(mockExperience.details[idx]);
    });
  });

  it('should be responsive (snapshot for different widths)', () => {
    (fixture.nativeElement as HTMLElement).style.width = '350px';
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
    (fixture.nativeElement as HTMLElement).style.width = '700px';
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
    (fixture.nativeElement as HTMLElement).style.width = '1024px';
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
}); 