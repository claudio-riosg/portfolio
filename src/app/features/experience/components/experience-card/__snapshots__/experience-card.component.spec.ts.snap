// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceCardComponent should be responsive (snapshot for different widths) 1`] = `
<div
  id="root3"
  style="width: 350px;"
>
  <div
    class="experience-card"
  >
    <div
      appscrollreveal=""
      class="experience-header scroll-reveal-hidden"
      style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
    >
      <h3
        class="company"
      >
        Test Company
      </h3>
      <div
        class="role-period"
      >
        <span
          class="role"
        >
          Test Title
        </span>
        <span
          class="period"
        >
          (2020 - 2022)
        </span>
      </div>
    </div>
    <p
      appscrollreveal=""
      class="summary scroll-reveal-hidden"
      style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
    >
       Test summary for the experience card. 
    </p>
    <ul
      class="details"
    >
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         First detail 
      </li>
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         Second detail 
      </li>
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         Third detail 
      </li>
      
    </ul>
  </div>
</div>
`;

exports[`ExperienceCardComponent should be responsive (snapshot for different widths) 2`] = `
<div
  id="root3"
  style="width: 700px;"
>
  <div
    class="experience-card"
  >
    <div
      appscrollreveal=""
      class="experience-header scroll-reveal-hidden"
      style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
    >
      <h3
        class="company"
      >
        Test Company
      </h3>
      <div
        class="role-period"
      >
        <span
          class="role"
        >
          Test Title
        </span>
        <span
          class="period"
        >
          (2020 - 2022)
        </span>
      </div>
    </div>
    <p
      appscrollreveal=""
      class="summary scroll-reveal-hidden"
      style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
    >
       Test summary for the experience card. 
    </p>
    <ul
      class="details"
    >
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         First detail 
      </li>
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         Second detail 
      </li>
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         Third detail 
      </li>
      
    </ul>
  </div>
</div>
`;

exports[`ExperienceCardComponent should be responsive (snapshot for different widths) 3`] = `
<div
  id="root3"
  style="width: 1024px;"
>
  <div
    class="experience-card"
  >
    <div
      appscrollreveal=""
      class="experience-header scroll-reveal-hidden"
      style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
    >
      <h3
        class="company"
      >
        Test Company
      </h3>
      <div
        class="role-period"
      >
        <span
          class="role"
        >
          Test Title
        </span>
        <span
          class="period"
        >
          (2020 - 2022)
        </span>
      </div>
    </div>
    <p
      appscrollreveal=""
      class="summary scroll-reveal-hidden"
      style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
    >
       Test summary for the experience card. 
    </p>
    <ul
      class="details"
    >
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         First detail 
      </li>
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         Second detail 
      </li>
      <li
        appscrollreveal=""
        class="scroll-reveal-hidden"
        style="opacity: 0; transform: translateY(30px) scale(0.95); filter: blur(2px); transition: all 0.8s cubic-bezier(0.25, 0.46, 0.45, 0.94);"
      >
         Third detail 
      </li>
      
    </ul>
  </div>
</div>
`;
